build/pipeline-ci.yaml

```diff
@@ -6,16 +6,33 @@ trigger:
 
 pr: none
 
-pool:
-  vmImage: 'windows-latest'
-  demands:
-  - msbuild
-  - visualstudio
-
 variables:
   BuildPlatform: 'any cpu'
   BuildConfiguration: 'debug'
 
-steps:  
-- template: template-ci-and-pr.yaml
 
+stages:
+- stage:
+  jobs:
+  - job: 'BuildAndTest' 
+    pool:
+      vmImage: 'windows-latest'
+      demands:
+      - msbuild
+      - visualstudio
+    steps:
+
+    - template: template-ci-and-pr.yaml
+
+  - job: 'RunDeviceAuthTests'
+    dependsOn: 'BuildAndTest'
+    pool:
+        name: DevEx Win8 Automation
+        demands: agent.name -equals MSIDWIN802
```

Comment:
Seems a bit fragile o use `agent.name`, especially given that there are 3 agents provisioned that can run these tests. Please configure to use the pool instead of a specific agent (e.g. https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/pools-queues?view=azure-devops&tabs=yaml%2Cbrowser)

Reply:
Creating a new pool for just one agent is a bit too much in my opinion. If we were to follow this going forward we would end up with many pools with only 1 agent in them. The only way agent.name will change is if someone removes the agent from the pool entirely. which only I and a few other people have the permission to do which is why using agent.name has never given us issues in the past.

Reply:
Not sure I understand what you mean by "creating a new pool". It looks to me that there already exists a pool with 3 agents that were configured for this scenario: 

![image](https://user-images.githubusercontent.com/12273384/81178115-9c1deb80-8f9f-11ea-8e3b-913d18a35104.png)

Reply:
My question is really is - why do you have to add ` demands: agent.name -equals MSIDWIN802` - and thus limit this test on one specific agent. Can't these tests run on any of those 3 agents?

Reply:
Discussed offline